src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+
+# Repeated capability tests
+    def test_markers_rep_cap(self, test_rfsg_device_session):
+        marker = test_rfsg_device_session.markers[0]
+        requested_terminal_name = '/Dev0/PXI_Trig0'
+        marker.exported_marker_event_output_terminal = requested_terminal_name
+        assert marker.exported_marker_event_output_terminal == requested_terminal_name
+
+    def test_script_triggers_rep_cap(self, test_rfsg_device_session):
+        trigger = test_rfsg_device_session.script_triggers[0]
+        requested_terminal_name = '/Dev0/PXI_Trig0'
+        trigger.exported_script_trigger_output_terminal = requested_terminal_name
+        assert trigger.exported_script_trigger_output_terminal == requested_terminal_name
+
+    def test_waveform_rep_cap(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.ARB_WAVEFORM)
+        waveform_data = np.full(1000, 1 + 0j, dtype=np.complex128)
+        test_rfsg_device_session.write_arb_waveform('mywaveform', waveform_data, False)
+        requested_waveform_iq_rate = 1e6
+        test_rfsg_device_session.waveform['mywaveform'].waveform_iq_rate = requested_waveform_iq_rate
+        assert test_rfsg_device_session.waveform['mywaveform'].waveform_iq_rate == requested_waveform_iq_rate
+
+    def test_deembedding_port_rep_cap(self, test_rfsg_device_session):
+        port = test_rfsg_device_session.deembedding_port['']
+        requested_deembedding_type = nirfsg.DeembeddingTypeAttrVals.SCALAR
+        port.deembedding_type = requested_deembedding_type
+        assert port.deembedding_type == requested_deembedding_type
+
+# Configuration methods related tests
+    def test_configure_power_and_frequency(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_rf(2e9, -5.0)
+        assert test_rfsg_device_session.power_level == -5.0
+        assert test_rfsg_device_session.frequency == 2e9
+
+    def test_configure_output_enabled(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_output_enabled(True)
+        assert test_rfsg_device_session.output_enabled is True
+        test_rfsg_device_session.configure_output_enabled(False)
+        assert test_rfsg_device_session.output_enabled is False
+
+    def test_configure_generation_mode(self, test_rfsg_device_session):
+        # Test CW mode
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.CW)
+        assert test_rfsg_device_session.generation_mode == nirfsg.GenerationMode.CW
+        # Test ARB_WAVEFORM mode
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.ARB_WAVEFORM)
+        assert test_rfsg_device_session.generation_mode == nirfsg.GenerationMode.ARB_WAVEFORM
+        # Test SCRIPT mode
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.SCRIPT)
+        assert test_rfsg_device_session.generation_mode == nirfsg.GenerationMode.SCRIPT
```

Comment:
I think you should delete the `configure_generation_mode` function, as per [our API Guidelines](https://github.com/ni/nimi-python/wiki/API-design-guidelines).

> * Don't include functions that are simple wrappers around setting/getting an attribute
> * Redundancy means confusion
> * Clients don't get type safety benefit that you get in C function anyway

I feel much more strongly about this one than `configure_digital_edge_script_trigger`.
It appears to add no value to the API.
It's actually faster to type `session.generation_mode = nirfsg.GenerationMode.CW` than to call the function.
Going by the examples, even your team seems to prefer the property.

Feel free to overrule me if it does something besides just set the property.

Reply:
In case of pushback, our design guidelines provide further guidance:

> * If there's debate on whether a function should be included or not, then don't include it
>   * Trivial to add if needed
>   * Hard to remove once added